Core.ApplicationServices/Contract/ItContractService.cs

```diff
@@ -221,18 +221,20 @@ public Maybe<OperationError> TransferPayments(int contractId, Guid targetUnitUui
 
         public Result<IEnumerable<DataProcessingRegistration>, OperationError> GetDataProcessingRegistrationsWhichCanBeAssigned(int id, string nameQuery, int pageSize)
         {
-            if (string.IsNullOrEmpty(nameQuery)) throw new ArgumentException($"{nameof(nameQuery)} must be defined");
             if (pageSize < 1) throw new ArgumentException($"{nameof(pageSize)} must be above 0");
 
             return WithReadAccess<IEnumerable<DataProcessingRegistration>>(id, contract =>
-                _contractDataProcessingRegistrationAssignmentService
-                    .GetApplicableDataProcessingRegistrations(contract)
-                    .ByPartOfName(nameQuery)
-                    .OrderBy(x => x.Id)
+            {
+                var query = _contractDataProcessingRegistrationAssignmentService
+                    .GetApplicableDataProcessingRegistrations(contract);
+
+                if (!string.IsNullOrEmpty(nameQuery)) query = query.ByPartOfName(nameQuery);
```

Comment:
Without the input param check, the null string exception will first be thrown in QueryByPartOfName